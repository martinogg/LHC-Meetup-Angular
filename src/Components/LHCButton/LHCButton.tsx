
import React, { Component, Children } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

interface IProps {
    onSelected(): void
    styles?: any
}

export default class LHCButton extends Component<IProps> {

    constructor(props: IProps) {

        super(props);
    }

    render() {

        return <TouchableOpacity style={[localStyles.container, this.props.styles]} onPress={() => this.props.onSelected()}>
            {this.props.children}
        </TouchableOpacity>
    }
}

const localStyles = StyleSheet.create({
    container: {
        margin: 10,
        backgroundColor: '#2980b6',
        paddingVertical: 15
    },
})