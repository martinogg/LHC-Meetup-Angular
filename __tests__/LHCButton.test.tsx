import { shallow } from 'enzyme';
import React from 'react';
import { View } from 'react-native';
import LHCButton from '../src/Components/LHCButton/LHCButton'
import renderer from 'react-test-renderer';

test('LHCButton snapshot', () => {

    const navigation = { navigate: jest.fn() };
    expect(renderer.create(<LHCButton onSelected={() => { }} />)).toMatchSnapshot();
})

test('LHCButton press', async () => {

    const callback = jest.fn()
    const wrapper = shallow(<View>
        <LHCButton onSelected={() => { callback() }} />
    </View>);

    await wrapper.find(LHCButton).first().simulate('selected')

    expect(callback).toBeCalledTimes(1)
})