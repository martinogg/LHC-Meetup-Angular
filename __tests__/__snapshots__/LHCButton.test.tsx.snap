// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LHCButton snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2980b6",
      "margin": 10,
      "opacity": 1,
      "paddingVertical": 15,
    }
  }
/>
`;
